public/scripts/userpage.js

```diff
@@ -64,9 +75,92 @@ friendlyPix.UserPage = class {
         this.followingContainer.hide();
         this.nbFollowingContainer.removeClass('is-active');
       });
+
+      // Event bindings for Privacy Consent Dialog
+      this.privacyDialogButton.click(() => this.showPrivacyDialog());
+      this.privacyDialogSave.click(() => this.savePrivacySettings());
+      this.allowDataProcessing.change(() => this.toggleSubmitStates());
     });
   }
 
+  /**
+   * Sets initial state of Privacy Dialog.
+   */
+  showPrivacyDialog() {
+    const userPromise = new Promise(function(resolve, reject) {
+      firebase.auth().onAuthStateChanged(function(user) {
+        if (user) {
+          resolve(user.uid);
+        } else {
+          reject();
+        }
+      });
+    }).then((uid) => {
+      this.userId = uid;
+      this.initializePrivacySettings(uid);
+      this.privacyDialog.get(0).showModal();
+    })
+  }
+
+  /**
+   * Disable the submit button for the privacy settings until data privacy
+   * policy is agreed to.
+   */
+  toggleSubmitStates() {
+    if (this.allowDataProcessing.is(':checked')) {
+      this.privacyDialogSave.removeAttr("disabled");
+    } else {
+      this.privacyDialogSave.attr("disabled", true);
+    }
+  }
+
+
+  /**
+   * Fetches previously saved privacy settings if they exist and
+   * enables the Submit button if user has consented to data processing.
+   */
+  initializePrivacySettings() {
+    if (this.savedPrivacySettings == null) {
+      friendlyPix.firebase.getPrivacySettings(this.userId).then(snapshot => {
+        this.savedPrivacySettings = snapshot.val();
+        if (this.savedPrivacySettings) {
+          if (this.savedPrivacySettings.data_processing) {
+            this.allowDataProcessing.prop('checked', true);
+            this.privacyDialogSave.removeAttr("disabled");
+          }
+          if (this.savedPrivacySettings.content) {
+            this.allowContent.prop("checked", true);
+          } else {
+            this.uploadButton.prop("disabled", true);
+            this.mobileUploadButton.prop("disabled", true);
+          }
+          if (this.savedPrivacySettings.social) {
+            this.allowSocial.prop("checked", true);
+          }
+        }
+      })
+    }
+  }
+
+  /**
+   * Saves new privacy settings and closes the privacy dialog.
+   */
+  savePrivacySettings() {
+    const settings = {
+      data_processing: this.allowDataProcessing.prop("checked"),
+      content: this.allowContent.prop("checked"),
+      social: this.allowSocial.prop("checked")
+    };
+    const uri = `/privacy/${this.userId}`;
+    this.database.ref(uri).set(settings).then(() => {
```

Comment:
Can yo umove this to the `firebase.js` file (trying to keep all Firebase specific requests in there)